genie-agent/src/main/java/com/netflix/genie/agent/execution/ExecutionContext.java

```diff
@@ -44,4 +48,60 @@
      * @throws RuntimeException if the agent id is already set
      */
     void setAgentId(@NotBlank final String agentId);
+
+    /**
+     * Set the job process once it has been launched.
+     *
+     * @param jobProcess a process handle for the children job
+     */
+    void setJobProcess(final Process jobProcess);
```

Comment:
Does this whole class need to be an interface? Seems to me that we're only ever going to have one implementation and it would save a lot of this boiler plate as you could just use Lombok `@Data`

Reply:
Just trying to make implementation classes package-local. If I remove the interface I need to make this public (and same for other classes with a single implementation for which I created an interface, for example all Services)

Reply:
I think that services should have interfaces for sure but this is effectively a DTO I don't see the value of the added maintenance of an interface. Making these kinds of classes public I think is fine.

Reply:
Public vs package+interface is debatable, but this guy is not a DTO.
It does have behavior, it's mutable, it is not serializable, It is not sent to a remote interface, it's a singleton.